### PR TITLE
Hexview feature parity 

### PR DESCRIPTION
**Pull request by DubiousDoggo**
## Description
- QHexDocument reads directly from the VGMFile
- Moved some functionality from QHexDocument and QHexCursor into QHexView 
- Selected events in the tree view are highlighted in the hexview, bold with red outline, and visa versa
- Updated font

## How Has This Been Tested?

- General UI stress testing

## Screenshots:
- Container selection from tree view
image

- Multi line selection
image

- Contiguous Area
image

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project. 
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

**Review by sykhro — CHANGES_REQUESTED** [review]
Looks very good overall, please remember to squash the fixup commits.

File: src/ui/qt/workarea/VGMFileTreeView.h [hunk 1]
```diff
@@ -43,6 +43,7 @@ class VGMFileTreeView : public QTreeWidget {
   ~VGMFileTreeView() override = default;
 
   void addVGMItem(VGMItem *item, VGMItem *parent, const std::wstring &name);
+  auto getTreeWidgetItem(VGMItem *vgm_item) { return m_items.at(vgm_item); };
```
[the remark this answers is not in the document]
Comment (DubiousDoggo) [reply src/ui/qt/workarea/VGMFileTreeView.h:46]: Assuming everything else is working correctly, the only way I can see it failing is if a null pointer gets passed somehow. Any exception should get caught by Qt and display a message, so at least then if it does fail it'll be a lot easier to debug.

**Review by sykhro — COMMENTED** [review]
The changes are fine but they should be squashed onto their original commits (commit message says "remove dead code" but actually does a whole lot more).

File: src/ui/qt/workarea/VGMFileTreeView.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (sykhro) [reply src/ui/qt/workarea/VGMFileTreeView.h:46]: That's fine for me, if that ever throws then it means there's a bug

**Pull request merged**